File: killrvideo-service-ratings/src/main/java/com/killrvideo/service/rating/dao/VideoRatingByUserDao.java
package com.killrvideo.service.rating.dao;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.Select;
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes;
import com.killrvideo.service.rating.dto.VideoRatingByUser;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

@Dao
public interface VideoRatingByUserDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    CompletableFuture<Void> insert(VideoRatingByUser videoRatingByUser);

    @Select
    @StatementAttributes(consistencyLevel = "LOCAL_ONE")
    CompletableFuture<Optional<VideoRatingByUser>> findUserRating(UUID videoId, UUID userid);
}

File: killrvideo-service-ratings/src/main/kotlin/com/killrvideo/service/rating/dao/VideoRatingByUserDao.kt
package com.killrvideo.service.rating.dao

import com.datastax.oss.driver.api.mapper.annotations.Dao
import com.datastax.oss.driver.api.mapper.annotations.Insert
import com.datastax.oss.driver.api.mapper.annotations.Select
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes
import com.killrvideo.service.rating.dto.VideoRatingByUser
import java.util.*
import java.util.concurrent.CompletableFuture

@Dao
interface VideoRatingByUserDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    fun insert(videoRatingByUser: VideoRatingByUser): CompletableFuture<VideoRatingByUser?>

    @Select
    @StatementAttributes(consistencyLevel = "LOCAL_ONE")
    fun findUserRating(videoId: UUID, userid: UUID): CompletableFuture<VideoRatingByUser?>
}
